sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/WithTimestamps.java

```diff
@@ -92,7 +92,11 @@ private WithTimestamps(SerializableFunction<T, Instant> fn, Duration allowedTime
    * duration that timestamps can be shifted backwards from the timestamp of the input element.
    *
    * @see DoFn#getAllowedTimestampSkew()
+   * @deprecated This method permits a elements to be timestamped behind the watermark. These
```

Comment:
Done.

Also deprecated `withAllowedTimestampSkew()`, and added a "CAUTION" in the class documentation.